static/app/components/profiling/flamegraphZoomView.tsx

```diff
@@ -782,4 +808,51 @@ const Canvas = styled('canvas')`
   position: absolute;
 `;
 
+// loosely based spreadsheet navigation
+const keyDirectionMap: Record<string, Direction> = {
+  ArrowUp: 'up',
+  ArrowDown: 'down',
+  ArrowLeft: 'left',
+  ArrowRight: 'right',
+  Tab: 'right',
+  'Shift+Tab': 'left',
+  Enter: 'down',
+  'Shift+Enter': 'up',
+  w: 'up',
+  s: 'down',
+  a: 'left',
+  d: 'right',
+};
+
+function handleFlamegraphKeyboardNavigation(
+  evt: KeyboardEvent,
+  currentFrame: FlamegraphFrame | undefined,
+  inverted: boolean = false
+) {
+  if (!currentFrame) {
+    return null;
+  }
+
+  const keyboardKey = !inverted
+    ? evt.key
+    : evt.key === 'ArrowUp'
+    ? 'ArrowDown'
+    : evt.key === 'ArrowDown'
+    ? 'ArrowUp'
+    : evt.key;
+
+  const key = evt.shiftKey ? `Shift+${keyboardKey}` : keyboardKey;
+  const direction = keyDirectionMap[key];
+  if (!direction) {
+    return currentFrame;
+  }
+  const nextSelection = selectNearestFrame(currentFrame, direction);
```

Comment:
If currentFrame is not falsy, then no matter what key you press, the least you will always get back is the current frame right?

Reply:
yeah, if we don't have a keybind set we no-op and return whatever the currentFrame is, if one is already selected.